plugins/hls-overloaded-record-dot-plugin/src/Ide/Plugin/OverloadedRecordDot.hs

```diff
@@ -158,89 +170,65 @@ descriptor :: Recorder (WithPriority Log) -> PluginId
-                case supportsResolve of
-                    Just False -> Just $ WorkspaceEdit
-                                    (Just (Map.singleton (fromNormalizedUri
-                                            (normalizedFilePathToUri nfp))
-                                            (edits (IntMap.lookup crsM crsD))))
-                                    Nothing Nothing
-                    _ -> Nothing
-            resolveData crsM =
-                case supportsResolve of
-                    Just True -> Just $ toJSON $ ORDRD (caDocId ^. L.uri) crsM
-                    _         -> Nothing
-            mkCodeAction crsD crsM  = InR CodeAction
+        let mkCodeAction (crsM, nse)  = InR CodeAction
                 { -- We pass the record selector to the title function, so that
                   -- we can have the name of the record selector in the title of
                   -- the codeAction. This allows the user can easily distinguish
                   -- between the different codeActions when using nested record
                   -- selectors, the disadvantage is we need to print out the
                   -- name of the record selector which will decrease performance
-                _title = mkCodeActionTitle exts crsM crsD
+                  _title = mkCodeActionTitle exts crsM nse
                 , _kind = Just CodeActionKind_RefactorRewrite
                 , _diagnostics = Nothing
                 , _isPreferred = Nothing
                 , _disabled = Nothing
-                , _edit = changes crsM crsD
+                , _edit = Nothing
                 , _command = Nothing
-                , _data_ = resolveData crsM
+                , _data_ = Just $ toJSON $ ORD pId $ toJSON $ ORDRD (caDocId ^. L.uri) crsM
                 }
-            actions = map (mkCodeAction crsDetails) (RangeMap.filterByRange caRange crsMap)
+            actions = map mkCodeAction (RangeMap.filterByRange caRange crsMap)
         pure $ InL actions
     where
-    mkCodeActionTitle :: [Extension] -> Int -> IntMap.IntMap RecordSelectorExpr-> Text
-    mkCodeActionTitle exts crsM crsD =
+    mkCodeActionTitle :: [Extension] -> Int -> HsExpr (GhcPass 'Renamed) -> Text
+    mkCodeActionTitle exts crsM se =
         if OverloadedRecordDot `elem` exts
             then title
             else title <> " (needs extension: OverloadedRecordDot)"
         where
-            title = "Convert `" <> name (IntMap.lookup crsM crsD) <> "` to record dot syntax"
-            name (Just (RecordSelectorExpr _ se _)) = printOutputable se
-            name _                                  = ""
+            title = "Convert `" <> printOutputable se <> "` to record dot syntax"
+
+mkWorkspaceEdit:: Uri -> Int -> IntMap.IntMap RecordSelectorExpr -> [Extension] -> NextPragmaInfo-> Maybe WorkspaceEdit
+mkWorkspaceEdit uri crsM crsD exts pragma =
+    Just $ WorkspaceEdit
+    { _changes = Just (Map.singleton uri (edits (IntMap.lookup crsM crsD)))
```

Comment:
I'd probably handle this by pushing the lookup outside of this function and get the caller to deal with the failure.

Reply:
Yes, this is important because we need to respond with a specific error. Completely overlooked it here though.